lib/node_modules/@stdlib/math/base/special/riemann-zeta/src/main.c

```diff
@@ -0,0 +1,813 @@
+*
+*     -   max deviation: \\(2.020\mbox{e-}18\\)
+*     -   expected error: \\(-2.020\mbox{e-}18\\)
+*     -   max error found (double): \\(3.994987\mbox{e-}17\\)
+*
+* -   \\(\[1,2\]\\)
+*
+*     -   max deviation: \\(9.007\mbox{e-}20\\)
+*     -   expected error: \\(9.007\mbox{e-}20\\)
+*
+* -   \\((2,4\]\\)
+*
+*     -   max deviation: \\(5.946\mbox{e-}22\\)
+*     -   expected error: \\(-5.946\mbox{e-}22\\)
+*
+* -   \\((4,7\]\\)
+*
+*     -   max deviation: \\(2.955\mbox{e-}17\\)
+*     -   expected error: \\(2.955\mbox{e-}17\\)
+*     -   max error found (double): \\(2.009135\mbox{e-}16\\)
+*
+* -   \\((7,15)\\)
+*
+*     -   max deviation: \\(7.117\mbox{e-}16\\)
+*     -   expected error: \\(7.117\mbox{e-}16\\)
+*     -   max error found (double): \\(9.387771\mbox{e-}16\\)
+*
+* -   \\(\[15,36)\\)
+*
+*     -   max error (in interpolated form): \\(1.668\mbox{e-}17\\)
+*     -   max error found (long double): \\(1.669714\mbox{e-}17\\)
+*
+* @param s    input value
+* @return     function value
+*
+* @example
+* double out = stdlib_base_zeta( 1.1 );
+* // returns ~10.584
+*/
+double stdlib_base_zeta( const double s ) {
+	double tmp;
+	double scc;
+	int32_t is;
+	int32_t as;
+	int32_t n;
+	double sc;
+	double r;
+
+	// Check for `NaN`:
+	if ( stdlib_base_is_nan( s ) ) {
+		return 0.0 / 0.0; // NaN
+	}
+
+	// Check for a pole:
+	if ( s == 1.0 ) {
+		return 0.0 / 0.0; // NaN
+	}
+
+	// Check for large value:
+	if ( s >= 56.0 ) {
+		return 1.0;
+	}
+
+	// Check for a closed form (integers):
+	if ( stdlib_base_is_integer( s ) ) {
+		// Cast `s` to a 32-bit signed integer:
+		is = (int32_t)s;
+
+		// Check that `s` does not exceed MAX_INT32:
+		if ( is == s ) {
+			if ( is < 0 ) {
+				as = ( -is );
+
+				// Check if even negative integer:
+				if ( ( as & 1 ) == 0 ) {
+					return 0.0;
+				}
+				n = ( ( as + 1 ) / 2 );
+
+				// Check if less than max Bernoulli number:
+				if ( n <= MAX_BERNOULLI_2N ) {
+					return -BERNOULLI[ n ] / ( as + 1 );
+				}
+
+				// Fall through...
+			} else if ( ( is & 1 ) == 0 ) { // Check if even nonnegative integer:
+				return EVEN_NONNEGATIVE_INTEGERS[ is / 2 ];
+			} else { // Must be a odd positive integer:
+				return ODD_POSITIVE_INTEGERS[ ( is - 3 ) / 2 ];
+			}
+		}
+		// Fall through...
+	}
+	if ( stdlib_base_abs( s ) < STDLIB_CONSTANT_FLOAT64_SQRT_EPS ) {
+		return -0.5 - ( STDLIB_CONSTANT_FLOAT64_LN_SQRT_TWO_PI * s );
+	}
+	scc = s;
+	sc = 1.0 - scc;
+	if ( scc < 0.0 ) {
+		// Check if even negative integer:
+		if ( stdlib_base_floor( scc / 2.0 ) == scc / 2.0 ) {
+			return 0.0;
+		}
+
+		// Swap `scc` and `sc`:
+		tmp = scc;
+		scc = sc;
+		sc = tmp;
+
+		// Determine if computation will overflow:
+		if ( scc > STDLIB_CONSTANT_FLOAT64_MAX_SAFE_NTH_FACTORIAL ) {
+			tmp = stdlib_base_sinpi( 0.5 * sc ) * 2.0 * stdlib_base_zeta( scc );
+			r = stdlib_base_gammaln( scc );
+			r -= scc * stdlib_base_ln( STDLIB_CONSTANT_FLOAT64_TWO_PI );
+			if ( r > MAX_LN ) {
```

Comment:
You made `MAX_LN` an `int32_t` above, but here we are performing double-precision comparison. Consider making `MAX_LN` a double.